hive.go

```diff
@@ -86,6 +86,18 @@ func mainInHost(daemon *docker.Client, overrides []string, cacher *buildCacher)
 	// Retrieve the versions of all clients being tested
 	if results.Clients, err = fetchClientVersions(daemon, *clientPattern, cacher); err != nil {
 		log15.Crit("failed to retrieve client versions", "error", err)
+		b, ok := err.(*buildError)
+		if ok {
+			results.Clients = make(map[string]map[string]string)
+			results.Clients[b.Client()] = make(map[string]string)
+			results.Clients[b.Client()] = map[string]string{"Failure": "Error building docker container for client"}
```

Comment:
This looks odd. `results.Clients[b.Client()] = make(map[string]string)` initializes and empty map at `results.Clients[b.Client()]`, which is then overwritten on the line after. 
IIUC youcna just remove L92

Reply:
I originally wrote it without line 91 and 92 but it gave an uninitialized map error. I can try removing line 92, but I think it will be the same uninitialized map error. I'll let you know what I find.